### PR TITLE
Refactor probes based on performance feedback

### PR DESCRIPTION
**Pull request by pega-talba**

### CODE REVIEW

**Review by misterdorito — APPROVED** [review]

**Review by taz-pega-work — APPROVED** [review]
excellent new content summaries. I see that you updated the percentages to integers for maxSurge and maxUnavailable in several yamls...that detail for what it means isn't documented anywhere. I makes me wonder if that should be a future enhancement for the effected helm charts in charts/pega. Nothing in your updates.

**Comment by APegaDavis** [comment]
> Note: [pegasystems/docker-pega-web-ready#119](https://github.com/pegasystems/docker-pega-web-ready/pull/119) is required before this change can be merged

Just wanted to note that we made the necessary changes as part of https://github.com/pegasystems/pega-helm-charts/pull/261 when adding a default server.xml to the config/deploy directory.

**Review by APegaDavis — APPROVED** [review]

**Comment by pegaautomationuser** [comment]
Starting PR-264 validation on -> integ-all

File: charts/pega/README.md [hunk 1]
```diff
@@ -393,17 +393,22 @@ tier:
...
-`successThreshold`    | Minimum consecutive successes for the probe to be considered successful after it determines a failure. | `1`
-`failureThreshold`    | The number consecutive failures for the pod to be terminated by Kubernetes. | `3`
+### Liveness, readiness, and startup probes
+
+[Probes are used by Kubernetes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) to determine application health.  Configure a probe for *liveness* to determine if a Pod has entered a broken state; configure it for *readiness* to determine if the application is available to be exposed; configure it for *startup* to determine if a pod is ready to be checked for liveness.  You can configure probes independently for each tier.  If not explicitly configured, default probes are used during the deployment.  Set the following parameters as part of a `livenessProbe`, `readinessProbe`, or `startupProbe` configuration.
+
+Notes:
+* `startupProbe` is only supported as of Kubernetes 1.18. If running a version older than 1.18, `startupProbe` will be ignored and different default values will be used for `livenessProbe` and `readinessProbe`.
```
Comment (taz-pega-work) [inline charts/pega/README.md:401]: Kubernetes 1.18 and later supports `startupProbe`. If your deployment uses a Kubernetes version older than 1.18, it ignores your `startupProbe` settings and uses different default values for `livenessProbe` and `readinessProbe`.

File: charts/pega/README.md [hunk 2]
```diff
@@ -393,17 +393,22 @@ tier:
...
+* `timeoutSeconds` cannot be greater than `periodSeconds` in some GCP environments. See [this API library from Google](https://developers.google.com/resources/api-libraries/documentation/compute/v1/csharp/latest/classGoogle_1_1Apis_1_1Compute_1_1v1_1_1Data_1_1HttpHealthCheck.html#a027a3932f0681df5f198613701a83145).
```
Comment (taz-pega-work) [inline charts/pega/README.md:402]: For details, see [this API library from Google]

File: charts/pega/README.md [hunk 3]
```diff
@@ -393,17 +393,22 @@ tier:
...
+* Default values are listed below in order of liveness, readiness, and startup.
+
+Parameter           | Description    | Default - 1.18+ | Default - pre 1.18
```
Comment (taz-pega-work) [inline charts/pega/README.md:405]: Default - pre-1.18

File: charts/pega/README.md [hunk 4]
```diff
@@ -393,17 +393,22 @@ tier:
...
-`periodSeconds`       | How often (in seconds) to perform the probe. Some providers such as GCP require this value to be greater than the timeout value. | `30`
-`successThreshold`    | Minimum consecutive successes for the probe to be considered successful after it determines a failure. | `1`
-`failureThreshold`    | The number consecutive failures for the pod to be terminated by Kubernetes. | `3`
+### Liveness, readiness, and startup probes
+
+[Probes are used by Kubernetes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) to determine application health.  Configure a probe for *liveness* to determine if a Pod has entered a broken state; configure it for *readiness* to determine if the application is available to be exposed; configure it for *startup* to determine if a pod is ready to be checked for liveness.  You can configure probes independently for each tier.  If not explicitly configured, default probes are used during the deployment.  Set the following parameters as part of a `livenessProbe`, `readinessProbe`, or `startupProbe` configuration.
```
Comment (taz-pega-work) [inline charts/pega/README.md:398]: Pega supports using liveness, readiness, and startup probes to determine application health in your deployments. For an overview of these probes, see [Configure Liveness, Readiness and Startup Probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) .

**Review by taz-pega-work — APPROVED** [review]
nice;-) I called out a few minor tweaks might help...up to you if you agree.

**Comment by pegaautomationuser** [comment]
Starting PR-264 validation on -> integ-all

File: charts/pega/README.md (hunk 1 above)
Comment (pega-talba) [reply charts/pega/README.md:401]: Done (with minor modifications)

File: charts/pega/README.md (hunk 2 above)
Comment (pega-talba) [reply charts/pega/README.md:402]: Done

File: charts/pega/README.md (hunk 3 above)
Comment (pega-talba) [reply charts/pega/README.md:405]: Done

File: charts/pega/README.md (hunk 4 above)
Comment (pega-talba) [reply charts/pega/README.md:398]: Done (with minor modifications)

**Comment by pegaautomationuser** [comment]
Starting PR-264 validation on -> integ-gke

**Comment by pegaautomationuser** [comment]
Starting PR-264 validation on -> integ-gke

**Pull request merged**